examples/loader-snowflake/src/data/snowflake.ts

```diff
@@ -0,0 +1,68 @@
+import "dotenv/config";
+import type {Binds, Connection, ConnectionOptions} from "snowflake-sdk";
+import snowflake from "snowflake-sdk";
+
+const {
+  SNOWFLAKE_ACCOUNT,
+  SNOWFLAKE_USERNAME,
+  SNOWFLAKE_PASSWORD,
+  SNOWFLAKE_DATABASE,
+  SNOWFLAKE_SCHEMA,
+  SNOWFLAKE_WAREHOUSE,
+  SNOWFLAKE_ROLE
+} = process.env;
+
+if (!SNOWFLAKE_ACCOUNT) throw new Error("missing SNOWFLAKE_ACCOUNT");
+if (!SNOWFLAKE_USERNAME) throw new Error("missing SNOWFLAKE_USERNAME");
+
+const options: ConnectionOptions = {
+  account: SNOWFLAKE_ACCOUNT,
+  username: SNOWFLAKE_USERNAME,
+  password: SNOWFLAKE_PASSWORD,
+  database: SNOWFLAKE_DATABASE,
+  schema: SNOWFLAKE_SCHEMA,
+  warehouse: SNOWFLAKE_WAREHOUSE,
+  role: SNOWFLAKE_ROLE
+};
+
+export async function run<T>(f: (query: (sql: string, params?: Binds) => Promise<any[]>) => Promise<T>): Promise<T> {
+  const connection = await connect(options);
+  try {
+    return await f((sql, params) => execute(connection, sql, params));
+  } finally {
+    await destroy(connection);
+  }
+}
+
+async function connect(options: ConnectionOptions): Promise<Connection> {
+  const connection = (snowflake as any).createConnection(options);
+  await new Promise<void>((resolve, reject) => {
+    connection.connect((error) => {
+      if (error) return reject(error);
+      resolve();
+    });
+  });
+  return connection;
```

Comment:
Is it worth creating a connection pool instead of a single connection? https://docs.snowflake.com/en/developer-guide/node-js/nodejs-driver-connect#label-create-connection-pool what's the expected concurrency behavior when people have multiple snowflake data loaders?

Reply:
I don’t think that’s necessary for this example, no. Users can always add that themselves if they want it, and I’d rather keep this example simple. In our experience, most data loaders issue a single query and only use a single connection. So there would be no benefit to having a connection pool. If you have multiple data loaders, then run in separate processes and wouldn’t be able to share a connection pool. Framework currently runs data loaders serially but we might parallelize in the future.